openedx/core/djangoapps/user_api/accounts/tests/pact/user-middleware.py

```diff
@@ -0,0 +1,32 @@
+"""
+Contain the middleware logic needed during pact verification
+"""
+from django.contrib import auth
+from django.utils.deprecation import MiddlewareMixin
+
+User = auth.get_user_model()
+
+
+class AuthenticationMiddleware(MiddlewareMixin):
+    """
+    Middleware to add default authentication into the requests for pact verification.
+
+    This middleware is required to add a default authenticated user and bypass CSRF validation
+    into the requests during the pact verification workflow. Without the authentication, the pact verification
+    process will not work as the apis.
+    See https://docs.pact.io/faq#how-do-i-test-oauth-or-other-security-headers
+    """
+
+    def __init__(self, get_response):
+        super().__init__(get_response)
+        self.auth_user = User.objects.get_or_create(username='staff', is_staff=True)[0]
```

Comment:
Can we switch to `get`?

Reply:
Yeah, shouldn't be a problem since we're creating the user beforehand.